test/AspNetCoreModule.Test/Framework/InitializeTestMachine.cs

```diff
@@ -65,27 +83,135 @@ public class InitializeTestMachine : IDisposable
         private static bool _InitializeTestMachineCompleted = false;
         private string _setupScriptPath = null;
         
-        private bool CheckPerquisiteForANCMTest()
+        private static bool? _MakeCertExeAvailable = null;
+        public static bool? MakeCertExeAvailable
         {
-            bool result = true;
-            TestUtility.LogInformation("CheckPerquisiteForANCMTest(): Environment.Is64BitOperatingSystem: {0}, Environment.Is64BitProcess {1}", Environment.Is64BitOperatingSystem, Environment.Is64BitProcess);
-            TestUtility.LogInformation("%ANCMTestFlags%: {0}", Environment.ExpandEnvironmentVariables(ANCMTestFlagsEnvironmentVariable));
+            get
+            {
+                if (_MakeCertExeAvailable == null)
+                {
+                    _MakeCertExeAvailable = false;
+                    try
+                    {
+                        string makecertExeFilePath = TestUtility.GetMakeCertPath();
+                        TestUtility.RunCommand(makecertExeFilePath, null, true, true);
+                        TestUtility.LogInformation("Verified makecert.exe is available : " + makecertExeFilePath);
+                        _MakeCertExeAvailable = true;
+                    }
+                    catch
+                    {
+                        _MakeCertExeAvailable = false;
+                    }
+                }
+                return _MakeCertExeAvailable;
+            }
+        }
 
-            if (Environment.Is64BitOperatingSystem && !Environment.Is64BitProcess)
+        public static string TestRootDirectory
+        {
+            get
             {
-                TestUtility.LogInformation("CheckPerquisiteForANCMTest() Failed: ANCM test should be started with x64 process mode on 64 bit machine; if you run this test on Visual Studio, you should set X64 first after selecting 'Test -> Test Settings -> Default Process Architecture' menu");
-                result = false;
+                return Path.Combine(Environment.ExpandEnvironmentVariables("%SystemDrive%") + @"\", "_ANCMTest");
             }
-            return result;
         }
+
+        private static string _globalTestFlags = null;
+        public static string GlobalTestFlags
+        {
+            get
+            {
+                if (_globalTestFlags == null)
+                {
+                    bool isElevated;
+                    WindowsIdentity identity = WindowsIdentity.GetCurrent();
```

Comment:
These lines can be condense significantly. Ex identity is only used on the next line, so just pass in WindowsIdentity.GetCurrent() into it.

Reply:
fixed